lib/chef/client.rb

```diff
@@ -638,13 +642,21 @@ def node_name
     # @api private
     #
     def register(client_name = node_name, config = Chef::Config)
+      puts "\n cli.rb - This is the client name I am using : #{client_name}"
+      puts "\n cli.rb - This is my client key : #{config[:client_key]}"
       if !config[:client_key]
         events.skipping_registration(client_name, config)
         logger.trace("Client key is unspecified - skipping registration")
-      elsif File.exists?(config[:client_key])
+      elsif detect_certificate_key(client_name)
```

Comment:
change this to `Chef::Http::Authenticator.detect_certificate_key(client_name)` and move it all to the other class and declare that as `def self.detect_certificate_key(client_name)`

Reply:
down in the authenticator class you'll probably want to create a method to delegate an instance method to the class method:

```
def detect_certificate_key(client_name)
  self.class.detect_certificate_key(client_name)
end

def self.detect_certificate_key(client_name)
[...]
end
```

that lets you call `detect_certificate_key(client_name)` from other methods in the authenticator class without having to remember to call the class method